fabric-datafixer-v1/src/main/java/net/fabricmc/fabric/impl/datafixer/FabricDataFixerImpl.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.datafixer;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.MessageFactory;
+import org.apache.logging.log4j.message.SimpleMessage;
+import com.google.common.base.Preconditions;
+import com.mojang.datafixers.DSL;
+import com.mojang.datafixers.DataFixUtils;
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.Dynamic;
+import com.mojang.datafixers.schemas.Schema;
+import com.mojang.datafixers.types.Type;
+import com.mojang.datafixers.types.templates.TypeTemplate;
+
+import net.minecraft.datafixers.DataFixTypes;
+import net.minecraft.datafixers.NbtOps;
+import net.minecraft.datafixers.Schemas;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.SharedConstants;
+
+import net.fabricmc.fabric.api.datafixer.v1.DataFixerEntrypoint;
+import net.fabricmc.fabric.api.datafixer.v1.DataFixerHelper;
+import net.fabricmc.loader.api.FabricLoader;
+
+public final class FabricDataFixerImpl implements DataFixerHelper {
+	public static final Logger LOGGER = LogManager.getLogger("Fabric-DataFixer", new MessageFactory() {
+		@Override
+		public Message newMessage(Object message) {
+			return new SimpleMessage("[Fabric-DataFixer] " + message);
+		}
+
+		@Override
+		public Message newMessage(String message) {
+			return new SimpleMessage("[Fabric-DataFixer] ");
+		}
+
+		@Override
+		public Message newMessage(String message, Object... params) {
+			return new SimpleMessage("[Fabric-DataFixer] " + params);
+		}
+	});
```

Comment:
uh what?

Reply:
Well that was a brainfart, I threw the wrong parameter into the simple message.